File: algorithmic_toolbox/week1_programming_challenges/2_maximum_pairwise_product/cpp/main.cpp
#include <iostream>
#include <vector>

#include "max_pairwise_product.hpp"

int main() {
  int n;
  std::cin >> n;
  std::vector<long long> numbers(n);
  for (int i = 0; i < n; ++i)
    std::cin >> numbers[i];

  long long result = maxPairwiseProduct(numbers);
  std::cout << result << std::endl;
  return 0;
}

File: algorithmic_toolbox/week1_programming_challenges/2_maximum_pairwise_product/go/main.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scanf("%d", &n)

	numbers := make([]int64, n)
	for i := 0; i < n; i++ {
		var num int64
		fmt.Scanf("%d", &num)
		numbers[i] = num
	}

	result := maxPairwiseProduct(numbers)
	fmt.Println(result)
}
